studio/components/grid/components/common/BlockKeys.tsx

```diff
@@ -33,7 +33,10 @@ export const BlockKeys: FC<BlockKeysProps> = ({ value, children, onEscape, onEnt
   )
 
   function onBlur() {
-    if (onEnter) onEnter(value)
+    // [Joshen] Commenting this out for now as its causing some odd behavior
+    // where its triggering when a children of the Popover overlay component is clicked
+    // Also to make the save action a bit more deliberate
+    // if (onEnter) onEnter(value)
```

Comment:
Probably best to do in another PR, but I think it would be nice to prevent the json editor from closing if there are unsaved changes and the user blurs it. I found myself making a few changes, and then accidentally clicking outside of the editor and losing them!

Reply:
gotcha will take a look into this one - its tricky cause the onBlur is getting triggered whenever BlockKey has a Button component rendered in it or when i click anywhere in this space
![image](https://user-images.githubusercontent.com/19742402/216957646-378b34d4-8799-4863-af2f-705f925f9d6b.png)

hence why i commented it out for now

Reply:
hmm okay this turns out to be a little more tricky because we're using the popover component which automatically closes once we click outside of it.

for now - i've found a workaround for the above issue i raised and i'll reinstate the previous behaviour where losing focus of the editor will save the changes